src/TLuaInterpreter.cpp

```diff
@@ -104,6 +104,69 @@ static bool isMain(const QString& name)
     return false;
 }
 
+static const char *bad_window_type = "%s: bad argument #%d type (window name as string expected, got %s)!";
+static const char *bad_cmdline_type = "%s: bad argument #%d type (command line name as string expected, got %s)!";
+static const char *bad_window_value = "window \"%s\" not found";
+static const char *bad_cmdline_value = "command line \"%s\" not found";
+
+#define WINDOW_NAME(_L, _pos)                                                                  \
+    ({                                                                                         \
+        int pos = (_pos);                                                                      \
```

Comment:
Sure. If "_pos" expands to "++s", as it does in a few calls to this macro, we want to evaluate it **once**.

Reply:
Also in case it's a more complicated expression, the parentheses make sure that it's evaluated as an expression and not something else.

Reply:
I though '_' prefixed variables are reserved in the global file scope: https://stackoverflow.com/a/20496955/4805858 .

Reply:
`_pos` is not a variable, it's a macro parameter. There's no way for that to collide with an external variable, macro definition, or whatever, because these get masked by the parameter.
In fact the opposite is true: precisely *because* `_pos` shall not be used as a variable we know that the expansion here is safe, as you're not allowed to pass in something named "_pos" or expanding to "_pos". (Even that would normally not be a problem, but you can play interesting games with C macros …)